src/schema/lastmod.go

```diff
@@ -1,40 +1,145 @@
 package schema
 
 import (
-	"io/ioutil"
+	"errors"
+	"fmt"
+	"io/fs"
 	"os"
+	"path/filepath"
+	"sort"
 	"time"
 
+	"github.com/uoregon-libraries/gopkg/fileutil"
 	"github.com/uoregon-libraries/newspaper-curation-app/src/internal/logger"
 )
 
+const manifestFilename = ".manifest"
+
+type fileInfo struct {
+	Path     string
+	Size     int64
+	Checksum string
+}
+
+func newFileInfo(loc string, e os.DirEntry) (fileInfo, error) {
+	var fd = fileInfo{Path: filepath.Join(loc, e.Name())}
+	var info, err = e.Info()
+	if err != nil {
+		return fd, fmt.Errorf("reading info for %q: %w", fd.Path, err)
+	}
+
+	fd.Size = info.Size()
+	fd.Checksum, err = fileutil.CRC32(fd.Path)
+	if err != nil {
+		return fd, fmt.Errorf("crc32 for %q: %w", fd.Path, err)
+	}
+
+	return fd, nil
+}
+
+type manifest struct {
+	Path    string
+	Created time.Time
+	Files   []fileInfo
+}
+
+func newManifest(location string) *manifest {
+	return &manifest{Path: location, Created: time.Now()}
+}
+
+func (m *manifest) build() error {
+	var entries, err = os.ReadDir(m.Path)
+	if err != nil {
+		return fmt.Errorf("reading dir %q: %s", m.Path, err)
```

Comment:
It's generally not a big deal because unwrapping errors is something I almost never bother to do, but yes, it's probably almost always better to use `%w`. In this case, just a stupid omission on my part :)

Reply:
I think making error handling consistent in using `%w` across the app is out of scope for this PR anyway after seeing how frequently `fmt.Errorf` is used is. Should we make a low priority (i.e. probably not worth including on the grant project board) issue for this though?

Reply:
App-wide, it's a good idea to get into the habit, but it doesn't matter since the code isn't meant to be used outside NCA.

If you want to make a ticket, I'll leave that up to you... but honestly I don't know when I'd choose to fix that compared to other issues (unless some of this becomes turned into consumable packages).

Reply:
Hmm. I'll make the issue just in case either of us is looking for something less complex to work on quick before the end of a day or after a mentally taxing refactor. And it'll be a reminder in case we fall out of the habit.

https://github.com/uoregon-libraries/newspaper-curation-app/issues/176